client/src/featureform/providers/filestore.py

```diff
@@ -0,0 +1,139 @@
+        # create an array with new values to be inserted then process them
+        # all at the same time in memory.
+        self.flush()
+        df = self.df
+        try:
+            value = df[df["entity"] == key]["value"].astype(self.stype).values[0]
+        except IndexError:
+            raise ValueError(
+                f"Could not find value for feature: {self.name} ({self.variant}) for: {key}"
+            )
+        if self.stype == "datetime64[ns]":
+            return pd.to_datetime(value)
+        return value
+
+    def flush(self):
+        self.df = (
+            pd.concat([self.df, self.update_df])
+            .drop_duplicates("entity", keep="last")
+            .sort_values("entity", ascending=False)
+            .reset_index(drop=True)
+        )
+        self.df.to_csv(self.filepath, mode="w", index=False, header=True)
+
+
+class LocalFileStore(OnlineStore):
+    def __init__(self):
+        self.path = ".featureform/features"
+        self.type_table = f"{self.path}/types.csv"
+        if not os.path.exists(self.path):
+            os.makedirs(self.path, exist_ok=True)
+
+    def create_table(self, name, variant, entity_type: ScalarType):
+        if os.path.exists(f"{self.path}/{name}/{variant}.csv"):
+            raise ValueError(f"Table {name} with variant {variant} already exists")
+
+        self.__set_type(name, variant, entity_type)
+
+        table = LocalFileTable(
+            name, variant, self.path, self.__convert_type(entity_type)
+        )
+        return table
+
+    def __set_type(self, name, variant, entity_type: ScalarType):
+        header = True
+        if os.path.exists(self.type_table):
+            header = False
+
+        types_df = pd.DataFrame(
+            data={"name": [name], "variant": [variant], "type": [entity_type.value]}
+        ).astype({"name": "string", "variant": "string", "type": "string"})
+
+        types_df.to_csv(self.type_table, mode="a", header=header, index=False)
+
+    def get_table(self, name, variant):
+        if not os.path.exists(f"{self.path}/{name}/{variant}.csv"):
+            raise ValueError(f"Table {name} with variant {variant} does not exist")
+
+        stype = self.__get_type(name, variant)
+
+        return LocalFileTable(name, variant, self.path, self.__convert_type(stype))
+
+    def __get_type(self, name, variant) -> ScalarType:
+        df = pd.read_csv(self.type_table)
+        matched_rows = df.loc[
+            (df["name"] == str(name)) & (df["variant"] == str(variant))
+        ]
+        if len(matched_rows) == 0:
+            raise ValueError(
+                f"Could not get type for table {name} with variant {variant}"
+            )
+
+        return ScalarType(matched_rows["type"].values[0])
+
+    def table_exists(self, name, variant):
+        return os.path.exists(f"{self.path}/{name}/{variant}.csv")
+
+    def is_vector_store(self):
+        return False
+
+    def __convert_type(self, stype: ScalarType):
```

Comment:
do we need this method?

Reply:
Yeah this converts featureform types to pandas types. we want to be strict on what gets converted to what